packages/dart_frog_auth/lib/src/dart_frog_auth.dart

```diff
@@ -0,0 +1,104 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:dart_frog/dart_frog.dart';
+
+extension on Map<String, String> {
+  String? authorization(String type) {
+    final value = this['Authorization']?.split(' ');
+
+    if (value != null && value.length == 2 && value.first == type) {
+      return value.last;
+    }
+
+    return null;
+  }
+
+  String? bearer() => authorization('Bearer');
+  String? basic() => authorization('Basic');
```

Comment:
Some documentation, to make my life easier when re-reading this code in a few months.

```suggestion
  /// Deconstructs the request header `Authorization` assuming it is using a
  /// `Basic` scheme.
  ///
  /// Returns a base64 encoded string of the format: `<username>:<password>`, if
  /// it exists, otherwise returns null.
  String? basic() => authorization('Basic');

  /// Deconstructs the request header `Authorization` assuming it is using a
  /// `Basic` scheme.
  ///
  /// Returns the token if it exists, otherwise returns null.
  String? bearer() => authorization('Bearer');
```

Reply:
This is just a private, helper extension used by this file to make it easier to parse the headers.

The middleware itself have a fairly good explanation on how the headers are structured, so I feel that adding that same docs on a private method, that will not be expose to the user, seems unnecessary.

Let me know if you feel too strong about though.

Reply:
I am gonna keep it as is, I do believe that this code is super simple and adding more docs to it will "bloat" it a bit.

If this causes issues in the future, we can revisit :)